src/inmanta/server/agentmanager.py

```diff
@@ -636,23 +637,41 @@ async def trigger_agent(self, env: UUID, agent_id: str) -> Apireturn:
         raise NotImplementedError()
 
     @protocol.handle(methods.list_agent_processes)
-    async def list_agent_processes(self, environment: Optional[UUID], expired: bool) -> Apireturn:
+    async def list_agent_processes(
+        self,
+        environment: Optional[UUID] = None,
+        expired: Optional[bool] = None,
+        start: Optional[UUID] = None,
+        end: Optional[UUID] = None,
+        limit: Optional[int] = None,
+    ) -> Apireturn:
+        query: Dict[str, Any] = {}
+        argscount = len([x for x in [start, end, limit] if x is not None])
+        if argscount == 3:
+            return 500, {"message": "Limit, start and end can not be set together"}
         if environment is not None:
+            query["environment"] = environment
             env = await data.Environment.get_by_id(environment)
             if env is None:
                 return 404, {"message": "The given environment id does not exist!"}
-
-        tid = environment
-        if tid is not None:
-            if expired:
-                aps = await data.AgentProcess.get_by_env(tid)
-            else:
-                aps = await data.AgentProcess.get_live_by_env(tid)
-        else:
-            if expired:
-                aps = await data.AgentProcess.get_list()
-            else:
-                aps = await data.AgentProcess.get_live()
+        if expired is None or not expired:
+            query["expired"] = None
```

Comment:
Kind of did it in https://github.com/inmanta/inmanta/pull/2521/commits/8115141e942dc11443a6149c5b79a845a68f08d5

But there is no expired agent process to actually filter, is this a problem?

Reply:
The problem is that if you set `expired=True`, you will also get non-expired agents in the result. That is very counter intuitive and this will lead to bugs. 

* If you only want to support the case where you want to retrieve only the non-expired agent processes, you should rename the argument from `expired` to `exclude_expired` or something similar. This will make it more clear
* If you want to support the both retrieving only expired and only non-expired processes, you should adjust the where  clause for the situation where `expired=True` to make sure it only returns expired processes.

Reply:
I know, `expired` should rather have been named `show_expired` to illustrate the current behavior, but I have no hands on this, this was already there and it has to stay backward compatible

Reply:
I see. That's a shame. I that case document that argument very will. Currently the documentation is not in-line with the implementation.

Reply:
Did it in https://github.com/inmanta/inmanta/pull/2521/commits/3dbee52a89786e96bae30b0e72916329da0897f2